src/upgrade.coffee

```diff
@@ -96,18 +97,37 @@ class Upgrade extends Command
           latestVersion = version if semver.gt(version, latestVersion)
 
         if latestVersion isnt pack.version and @hasRepo(pack)
-          callback(null, {pack, latestVersion})
+          callback(null, latestVersion)
         else
           callback()
 
+  getLatestSha: (pack, callback) ->
+    repoPath = path.join(@atomPackagesDirectory, pack.name)
+    @spawn 'git', ['fetch', 'origin', 'master'], {cwd: repoPath}, (code, stderr='', stdout='') ->
```

Comment:
@kuychaco We should extract `Develop::cloneRepository` into `Command` or some other class and use it here (and in `Install`) instead of shelling out ourselves, since it has the `config.getSetting 'git' ... git.addGitToEnv(process.env)` that is required to make it work on Windows.